[2024-10-13T23:31:59.737+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-10-13T23:31:59.751+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Nikita_Lapshin_experiment.train_and_log_model manual__2024-10-13T23:27:10.051779+00:00 [queued]>
[2024-10-13T23:31:59.756+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Nikita_Lapshin_experiment.train_and_log_model manual__2024-10-13T23:27:10.051779+00:00 [queued]>
[2024-10-13T23:31:59.756+0000] {taskinstance.py:2306} INFO - Starting attempt 3 of 4
[2024-10-13T23:31:59.767+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): train_and_log_model> on 2024-10-13 23:27:10.051779+00:00
[2024-10-13T23:31:59.782+0000] {standard_task_runner.py:64} INFO - Started process 869 to run task
[2024-10-13T23:31:59.797+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Nikita_Lapshin_experiment', 'train_and_log_model', 'manual__2024-10-13T23:27:10.051779+00:00', '--job-id', '140', '--raw', '--subdir', 'DAGS_FOLDER/HW2.py', '--cfg-path', '/tmp/tmpqqmdnl9w']
[2024-10-13T23:31:59.810+0000] {standard_task_runner.py:91} INFO - Job 140: Subtask train_and_log_model
[2024-10-13T23:31:59.932+0000] {task_command.py:426} INFO - Running <TaskInstance: Nikita_Lapshin_experiment.train_and_log_model manual__2024-10-13T23:27:10.051779+00:00 [running]> on host e332dadfa0c1
[2024-10-13T23:32:00.024+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Lapshin Nikita' AIRFLOW_CTX_DAG_ID='Nikita_Lapshin_experiment' AIRFLOW_CTX_TASK_ID='train_and_log_model' AIRFLOW_CTX_EXECUTION_DATE='2024-10-13T23:27:10.051779+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-13T23:27:10.051779+00:00'
[2024-10-13T23:32:00.025+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-10-13T23:32:00.052+0000] {s3.py:1422} INFO - Downloading source S3 file from Bucket mlops-2 with path datasets/prepared_data.csv
[2024-10-13T23:32:00.075+0000] {base.py:84} INFO - Using connection ID 's3_connection' for task execution.
[2024-10-13T23:32:00.077+0000] {connection_wrapper.py:388} INFO - AWS Connection (conn_id='s3_connection', conn_type='aws') credentials retrieved from login and password.
[2024-10-13T23:32:02.001+0000] {logging_mixin.py:188} INFO - result1
[2024-10-13T23:32:02.094+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:32:02 WARNING mlflow.utils.git_utils: Failed to import Git (the Git executable is probably not on your PATH), so Git SHA is not available. Error: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh(<full-path-to-git-executable>)

All git commands will error until this is rectified.

This initial message can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|silent|none|n|0: for no message or exception
    - warn|w|warning|log|l|1: for a warning message (logging level CRITICAL, displayed by default)
    - error|e|exception|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet
[2024-10-13T23:32:23.972+0000] {logging_mixin.py:188} INFO - result2 random_forest
[2024-10-13T23:32:30.333+0000] {credentials.py:1147} INFO - Found credentials in environment variables.
[2024-10-13T23:32:51.558+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:32:51 WARNING mlflow.models.model: Input example should be provided to infer model signature if the model signature is not provided when logging the model.
[2024-10-13T23:32:57.347+0000] {connectionpool.py:331} WARNING - Connection pool is full, discarding connection: storage.yandexcloud.net. Connection pool size: 10
[2024-10-13T23:32:57.758+0000] {connectionpool.py:331} WARNING - Connection pool is full, discarding connection: storage.yandexcloud.net. Connection pool size: 10
[2024-10-13T23:32:58.616+0000] {connectionpool.py:331} WARNING - Connection pool is full, discarding connection: storage.yandexcloud.net. Connection pool size: 10
[2024-10-13T23:33:03.825+0000] {connectionpool.py:331} WARNING - Connection pool is full, discarding connection: storage.yandexcloud.net. Connection pool size: 10
[2024-10-13T23:33:05.567+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:05 INFO mlflow.models.evaluation.default_evaluator: Computing model predictions.
[2024-10-13T23:33:06.194+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.10/site-packages/sklearn/metrics/_regression.py:492 FutureWarning: 'squared' is deprecated in version 1.4 and will be removed in 1.6. To calculate the root mean squared error, use the function'root_mean_squared_error'.
[2024-10-13T23:33:06.200+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:06 INFO mlflow.models.evaluation.default_evaluator: Testing metrics on first row...
[2024-10-13T23:33:06.202+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:06 WARNING mlflow.models.evaluation.default_evaluator: SHAP or matplotlib package is not installed, so model explainability insights will not be logged.
[2024-10-13T23:33:33.491+0000] {logging_mixin.py:188} INFO - result3 random_forest
[2024-10-13T23:33:33.554+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:33 INFO mlflow.tracking._tracking_service.client: üèÉ View run random_forest_run at: http://mlflow-service:5000/#/experiments/889983933329739548/runs/d4b00a8d0da54b3e97e798d7079cf8eb.
[2024-10-13T23:33:33.554+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:33 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow-service:5000/#/experiments/889983933329739548.
[2024-10-13T23:33:33.621+0000] {logging_mixin.py:188} INFO - result2 linear_regression
[2024-10-13T23:33:36.154+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.10/site-packages/_distutils_hack/__init__.py:18 UserWarning: Distutils was imported before Setuptools, but importing Setuptools also replaces the `distutils` module in `sys.modules`. This may lead to undesirable behaviors or errors. To avoid these issues, avoid using distutils directly, ensure that setuptools is installed in the traditional way (e.g. not an editable install), and/or make sure that setuptools is always imported before distutils.
[2024-10-13T23:33:36.157+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.10/site-packages/_distutils_hack/__init__.py:33 UserWarning: Setuptools is replacing distutils.
[2024-10-13T23:33:38.079+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:38 WARNING mlflow.models.model: Input example should be provided to infer model signature if the model signature is not provided when logging the model.
[2024-10-13T23:33:39.264+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:39 INFO mlflow.models.evaluation.default_evaluator: Computing model predictions.
[2024-10-13T23:33:39.295+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.10/site-packages/sklearn/metrics/_regression.py:492 FutureWarning: 'squared' is deprecated in version 1.4 and will be removed in 1.6. To calculate the root mean squared error, use the function'root_mean_squared_error'.
[2024-10-13T23:33:39.322+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:39 INFO mlflow.models.evaluation.default_evaluator: Testing metrics on first row...
[2024-10-13T23:33:39.333+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:39 WARNING mlflow.models.evaluation.default_evaluator: SHAP or matplotlib package is not installed, so model explainability insights will not be logged.
[2024-10-13T23:33:39.685+0000] {logging_mixin.py:188} INFO - result3 linear_regression
[2024-10-13T23:33:39.696+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:39 INFO mlflow.tracking._tracking_service.client: üèÉ View run linear_regression_run at: http://mlflow-service:5000/#/experiments/889983933329739548/runs/af24d30e530a48d1ba769bab342aa320.
[2024-10-13T23:33:39.697+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:39 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow-service:5000/#/experiments/889983933329739548.
[2024-10-13T23:33:39.972+0000] {logging_mixin.py:188} INFO - result2 decision_tree
[2024-10-13T23:33:43.253+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.10/site-packages/_distutils_hack/__init__.py:18 UserWarning: Distutils was imported before Setuptools, but importing Setuptools also replaces the `distutils` module in `sys.modules`. This may lead to undesirable behaviors or errors. To avoid these issues, avoid using distutils directly, ensure that setuptools is installed in the traditional way (e.g. not an editable install), and/or make sure that setuptools is always imported before distutils.
[2024-10-13T23:33:43.256+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.10/site-packages/_distutils_hack/__init__.py:33 UserWarning: Setuptools is replacing distutils.
[2024-10-13T23:33:45.014+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:45 WARNING mlflow.models.model: Input example should be provided to infer model signature if the model signature is not provided when logging the model.
[2024-10-13T23:33:47.234+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:47 INFO mlflow.models.evaluation.default_evaluator: Computing model predictions.
[2024-10-13T23:33:47.290+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.10/site-packages/sklearn/metrics/_regression.py:492 FutureWarning: 'squared' is deprecated in version 1.4 and will be removed in 1.6. To calculate the root mean squared error, use the function'root_mean_squared_error'.
[2024-10-13T23:33:47.305+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:47 INFO mlflow.models.evaluation.default_evaluator: Testing metrics on first row...
[2024-10-13T23:33:47.310+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:47 WARNING mlflow.models.evaluation.default_evaluator: SHAP or matplotlib package is not installed, so model explainability insights will not be logged.
[2024-10-13T23:33:50.669+0000] {logging_mixin.py:188} INFO - result3 decision_tree
[2024-10-13T23:33:50.720+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:50 INFO mlflow.tracking._tracking_service.client: üèÉ View run decision_tree_run at: http://mlflow-service:5000/#/experiments/889983933329739548/runs/adc05ec3a9e24cc68c6d88f6607b9508.
[2024-10-13T23:33:50.721+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:50 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow-service:5000/#/experiments/889983933329739548.
[2024-10-13T23:33:50.753+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:50 INFO mlflow.tracking._tracking_service.client: üèÉ View run @GodSiemens at: http://mlflow-service:5000/#/experiments/889983933329739548/runs/33f02aa5b0de42708d94d7b722192b6b.
[2024-10-13T23:33:50.754+0000] {logging_mixin.py:188} WARNING - 2024/10/13 23:33:50 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow-service:5000/#/experiments/889983933329739548.
[2024-10-13T23:33:50.811+0000] {python.py:237} INFO - Done. Returned value was: {'random_forest': {'start_time': 1728862321.8550305, 'end_time': 1728862386.2396758, 'model_score': 0.9996198031952006}, 'linear_regression': {'start_time': 1728862321.8550305, 'end_time': 1728862419.3901637, 'model_score': 0.8912486221145098}, 'decision_tree': {'start_time': 1728862321.8550305, 'end_time': 1728862427.345311, 'model_score': 1.0}}
[2024-10-13T23:33:50.813+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-10-13T23:33:50.898+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=Nikita_Lapshin_experiment, task_id=train_and_log_model, run_id=manual__2024-10-13T23:27:10.051779+00:00, execution_date=20241013T232710, start_date=20241013T233159, end_date=20241013T233350
[2024-10-13T23:33:50.984+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 0
[2024-10-13T23:33:51.058+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-13T23:33:51.060+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
